server/server.go

```diff
@@ -86,9 +188,35 @@ func (s *Server) Run() error {
 	return <-errChan
 }
 
+func (s *server) run() error {
+	return s.httpServer.Serve(s.httpListener)
+}
+
 func (s *Server) Shutdown() {
-	ctx, cancel := context.WithTimeout(context.Background(), s.cfg.ServerGracefulShutdownTimeout)
+	s.authServer.shutdown(s.cfg.ServerGracefulShutdownTimeout)
+	s.unAuthServer.shutdown(s.cfg.ServerGracefulShutdownTimeout)
+}
+
+func (s *server) shutdown(gracefulShutdownTimeout time.Duration) {
+	ctx, cancel := context.WithTimeout(context.Background(), gracefulShutdownTimeout)
 	defer cancel()
 
-	s.HTTPServer.Shutdown(ctx)
+	s.httpListener.Close()
+	s.httpServer.Shutdown(ctx)
+}
+
+func registerEndpoints(srv *server, reg *prometheus.Registry) {
+	srv.http.Handle("/metrics", promhttp.HandlerFor(reg, promhttp.HandlerOpts{}))
+	srv.http.Handle("/ready", http.HandlerFunc(readyHandler))
+	srv.http.Handle("/debug/pprof/", http.HandlerFunc(pprof.Index))
+}
+
+func readyHandler(w http.ResponseWriter, r *http.Request) {
+	w.WriteHeader(200)
```

Comment:
Not exactly. we need to only say we are ready if have started listening on both ports. It's tricky.
Don't fix it in this PR, leave it like it is now. Focus on metrics